### PR TITLE
Feature/network detection

### LINKED ISSUE #7045: v12: hide docker0 in list of Interfaces
**Describe the bug**
In configurator or in Interfaces menu, we can see `docker0`. It will let you think that you can edit this interface which is not the case. We should hide this interface from listing.

### LINKED ISSUE #7046: v12: changing hostname is not possible during configurator
**Describe the bug**
If you try to change hostname using configurator, API returns an error. However, if you run `hostnamectl` on PF server (not in container), hostname has been changed and is correctly set up at next reboot.

**To Reproduce**
1. Change hostname of your host through configurator

=> API return: `{"message":"Failed to change hostname to: linlog1","status":500}`

**Expected behavior**
API doesn't return an error.

**Additionnal context**
In logs, I can see that `api-frontend` handle request:

`Jun 29 11:56:47 linlog api-frontend-docker-wrapper[26437]: api-frontend-access 100.64.0.1 - - [29/Jun/2022:11:56:47 +0000] "PUT /api/v1/configurator/config/system/hostname HTTP/1.1" 500 64 "https://139.177.192.253:1443/admin" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36"`

but there is nothing obvious in logs except in syslog:

`Jun 29 11:53:29 linlog systemd-hostnamed[186313]: Changed static hostname to 'linlog1'
`

### LINKED ISSUE #7054: v12: /etc/network/interfaces is not rewritten
**Describe the bug**
On Debian, if you try to assign an IP address on an interface using web admin, it will work but IP address will be written in `/etc/network/interfaces` **inside** `pfperl-api` container in place of `/etc/network/interfaces` on host.

### PR DESCRIPTION
**Pull request by nqb**
# Description
* Handle network changes (interfaces, gateway, DNS and hostname) when changes are made inside containers
Routed networks and static routes are managed by keepalived (no containerized service).
* Hide docker interfaces in GUI

# Issues
fixes #7045
fixes #7046 
fixes #7054 

# Delete branch after merge
YES

### CODE REVIEW

**Comment by nqb** [comment]
Ready for review.

**Comment by jrouzierinverse** [comment]
What is the best way to test this?

File: lib/pf/UnifiedApi/Controller/Config/System.pm [hunk 1]
```diff
@@ -97,17 +96,30 @@ sub put_dns_servers {
     }
 }
 
+sub _get_hostname {
+    my ($self) = @_;
+    my $hostnamectl = pf_run("hostnamectl");
+    my $static_hostname;
+    for my $line (split(/\n/, $hostnamectl)) {
+	if($line =~ /^\s*Static hostname:\s*([a-zA-Z0-9]+)/) {
```
Comment (fdurand) [inline lib/pf/UnifiedApi/Controller/Config/System.pm]: Why don´t we just get directly what we have after "Static hostname:" ?

File: lib/pf/UnifiedApi/Controller/Config/System.pm [hunk 2]
```diff
@@ -97,17 +96,30 @@ sub put_dns_servers {
...
+	    $static_hostname = $1;
+	    last;
+	}
+    }
+    return $static_hostname;
```
Comment (fdurand) [inline lib/pf/UnifiedApi/Controller/Config/System.pm]: we should probably raise an error if $static_hostname is empty or null.

File: lib/pf/UnifiedApi/Controller/Config/System.pm [hunk 3]
```diff
@@ -97,17 +96,30 @@ sub put_dns_servers {
     }
 }
 
+sub _get_hostname {
+    my ($self) = @_;
+    my $hostnamectl = pf_run("hostnamectl");
```
Comment (jrouzierinverse) [inline lib/pf/UnifiedApi/Controller/Config/System.pm]: Use ```hostnamectl --static``` to avoid extra work

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by nqb** [comment]
> What is the best way to test this?

In unit tests you mean ?

**Comment by nqb** [comment]
@fdurand and @jrouzierinverse, I adjust code. Let me know if it's ok for you.

**Review by fdurand — APPROVED** [review]

**Pull request merged**